.github/workflows/ci.yaml

```diff
@@ -2,18 +2,36 @@ name: Cargo Tests
 
 on:
   pull_request:
+    paths:
```

Comment:
This paths filter adds maintenance cost. Every time you rename folders or add new areas that should trigger CI, someone has to remember to update the workflow or it silently doesn't run.